protostar/starknet_gateway/gateway_facade.py

```diff
@@ -292,8 +289,7 @@ async def declare_v0(
             result = await self._gateway_client.declare(tx, token)
             register_response(dataclasses.asdict(result))
             if wait_for_acceptance:
-                if self._logger:
-                    self._logger.info("Waiting for acceptance...")
+                logging.info("Waiting for acceptance...")
```

Comment:
We want to stop logging sometimes, right? That's why we didn't pass the logger higher up in some cases

Reply:
Was this an expected behaviour? From the code it looked to me rather like someone being lazy when implementing...